CMS-Enterprise/sbom-harbor
# feat(purl-generation): Syft Service now creates a well formed, sensible purl for the Package

## Summary

Syft service will generate a purl using the correct type instead of a cataloger

### Changed

1. The generate_stopgap_purl() function is renamed to generate_purl() and it now uses the CATALOGERS static hashmap to lookup the actual purl type from the cataloger used to generate the SBOM.
2. The function get_cataloger_to_build_target_map() has been removed and a BUILD_TARGETS static HashMap has been created to use instead.

## How to test

1. Set up test environment
2. Run GitHub Provider: `/target/debug/harbor ingest -p github harbor-test-org --debug`
3. Verify that all purls have types except `no-type`, which a discussion is planned to handle
4. Verify that all purls fit the purl specification: https://github.com/package-url/purl-spec
